### PR TITLE
grafana - add support for prometheus serviceMonitor

### PR DESCRIPTION
**Pull request by smcavallo**
add support for optional prometheus serviceMonitor

This is a pattern that is pretty standard across helm charts.  Some examples:
https://github.com/helm/charts/blob/master/stable/prometheus-redis-exporter/templates/servicemonitor.yaml
https://github.com/helm/charts/blob/master/stable/prometheus-cloudwatch-exporter/templates/servicemonitor.yaml
https://github.com/helm/charts/blob/master/stable/coredns/templates/servicemonitor.yaml
It would be helpful to have this built into the grafana helm chart as well for ease of monitoring.

Signed-off-by: S.Cavallo <smcavallo@hotmail.com>

### CODE REVIEW

File: charts/grafana/values.yaml [hunk 1]
```diff
@@ -124,6 +124,19 @@ service:
   labels: {}
   portName: service
 
+serviceMonitor:
+  ## If true, a ServiceMonitor CRD is created for a prometheus operator
+  ## https://github.com/coreos/prometheus-operator
+  ##
+  enabled: false
+  path: /metrics
+  #  namespace: monitoring
```
Comment (torstenwalter) [inline charts/grafana/values.yaml:133]: Can we set this to `namespace: ""` by default and maybe add a comment above what it does? Or which namespace is used by default.

**Comment by zanhsieh** [comment]
@smcavallo 
This PR need to rebase.

**Comment by smcavallo** [comment]
> 
> 
> @smcavallo
> This PR need to rebase.

rebased, thank you

File: charts/grafana/values.yaml (hunk 1 above)
Comment (smcavallo) [reply charts/grafana/values.yaml]: > 
> 
> Can we set this to `namespace: ""` by default and maybe add a comment above what it does? Or which namespace is used by default.

@torstenwalter - good catch, thank you.  I've updated the readme to reflect the additions to values.yaml with a description and default value.  If the documentation doesn't make sense please let me know and I can update the descriptions.

**Comment by torstenwalter** [comment]
Looks food to me, but I just helped to migrate this repository from stable/grafana. @zanhsieh @rtluckie @maorfr can one of you review?

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**